src/com/esotericsoftware/kryo/Kryo.java

```diff
@@ -417,9 +418,9 @@ public Registration register (Class type, Serializer serializer) {
 		return classResolver.register(new Registration(type, serializer, getNextRegistrationId()));
 	}
 
-	/** Registers the class using the specified ID and serializer. If the ID is already in use by the same type, the old entry is
-	 * overwritten. If the ID is already in use by a different type, a {@link KryoException} is thrown. Registering a primitive
-	 * also affects the corresponding primitive wrapper.
+	/** Registers the class using the specified ID and serializer. Providing an ID that is already in use by the same type
+	 * will cause the old entry to be overwritten. Providing an ID that is already in use by a different type is unsupported
+	 * and may cause undefined behavior. Registering a primitive also affects the corresponding primitive wrapper.

```

Comment:
Same here, I'd say that "Providing an ID that is already in use by a different type is unsupported and may cause undefined behavior." can be removed because it's already covered by the previous sentence "Providing an ID that is already in use by the same type will cause the old entry to be overwritten." (which is also correctly describing the implementation AFAICS).